### PR TITLE
Fixes X-Upstream-Hostname header for os hostnames with invalid characters.

### PR DESCRIPTION
**Pull request by chadjw**
Fixes X-Upstream-Hostname header for os hostnames with invalid characters. This occurs when during local development when your local computer name has an invalid character. 

For example, Apple like to name devices something like "Chad's MacBook Pro." This gets converted to chad's-macbook-pro as a hostname returned by the os.hostname() call. This change resolves the issue by removing the invalid characters.

### CODE REVIEW

File: src/middleware/headers.js [hunk 1]
```diff
@@ -62,7 +62,7 @@ module.exports = function (middleware) {
 		}
 
 		if (process.env.NODE_ENV === 'development') {
-			headers['X-Upstream-Hostname'] = os.hostname();
+			headers['X-Upstream-Hostname'] = os.hostname().replace(/[^0-9A-Za-z-]/g, '');
```
Comment (pitaj) [inline src/middleware/headers.js:65]: Should underscores and periods not be allowed?

**Comment by pitaj** [comment]
Are certain characters not allowed in http headers? I guess I'm wondering why an apostrophe, for instance, is such an issue in a non-standard header.

File: src/middleware/headers.js (hunk 1 above)
Comment (chadjw) [reply src/middleware/headers.js:65]: Chars based on https://man7.org/linux/man-pages/man7/hostname.7.html. Should have included a period as well (will be fixed in a new commit). Uppercase are technically not allowed either but didn't seem to break things.

**Comment by chadjw** [comment]
> Are certain characters not allowed in http headers? I guess I'm wondering why an apostrophe, for instance, is such an issue in a non-standard header.

Chars based on https://man7.org/linux/man-pages/man7/hostname.7.html. Should have included a period as well (was fixed in a latest commit). Uppercase are technically not allowed either but didn't seem to break things.

**Comment by pitaj** [comment]
That sounds more like a restriction for what is allowed in headers, in which case URL encoding might be a better solution.

**Comment by julianlam** [comment]
I have absolutely no idea why I even added this header 6 years ago, but this change looks ok to me even if not url-encoded. Thanks!

**Pull request merged**